src/utils/format.ts

```diff
@@ -15,27 +28,170 @@ function _getLocaleSymbols(): { thousands: string; decimals: string } {
 }
 
 const { thousands: THOUSANDS_SYMBOL, decimals: DECIMALS_SYMBOL } = _getLocaleSymbols()
-const DEFAULT_THOUSANDS_SYMBOL = ','
-const DEFAULT_DECIMALS_SYMBOL = '.'
+
+const _formatDecimalsFromWeiForDisplay = (decimalsToConvert: BN, decimalSymbol: string = DEFAULT_DECIMALS_SYMBOL) => '0' + decimalSymbol + fromWei(decimalsToConvert).slice(2)
+
+interface CompactionOptions {
+  precision: number,
+  thousandsSymbol: string,
+  decimalSymbol: string
+}
+
+function _compactLargeBNNumberToString(
+  baseBN: BN,
+  representationBN: BN,
+  {
+    precision,
+    thousandsSymbol,
+    decimalSymbol,
+  }: CompactionOptions): string {
+  // e.g TRILLION_123_123.div(ONE_TRILLION) = 123123.123123123
+  const numAsFraction = baseBN.mul(TEN.pow(new BN(precision))).div(representationBN)
+  const { integerPart, decimalPart } = _decomposeBn(numAsFraction, DEFAULT_LARGE_NUMBER_PRECISION, DEFAULT_LARGE_NUMBER_PRECISION)
+  // 123123.123123123 = 123,123.123123123
+  const formattedInteger = _formatNumber(integerPart.toString(), thousandsSymbol)
+  // no relevant decimal section
+  if (decimalPart.isZero()) return formattedInteger
+
+  const formattedDecimal = decimalPart
+    .toString()
+    .padStart(precision, '0') // Pad the decimal part with leading zeros
+    .replace(/0+$/, '') // Remove the right zeros
```

Comment:
Can `decimalPart` be `0`? Then `.replace(/0+$/, '')` will produce an empty string

Reply:
not sure

Reply:
oops this responded to wrong comment, it is fine as there is logic to catch 0 decimals in other places (just ignore it and it will go away)